src/_posts/2019-02-07-working-with-dynamodb.md

```diff
@@ -0,0 +1,995 @@
+
+First, we extract an `interface` from our repository with the method `save`. 
+
+```kotlin
+interface TaskRepository {
+    fun save(task: Task)
+}
+```
+
+Let's make a test for our repository. Starting our test we are going to need to connect to the database and create the table before doing any testing. 
+
+```kotlin
+class DynamoDBTaskRepositoryShould {
+
+    @Test
+    internal fun `add Task to table`() {
+        val endpoint = "http://localhost:8000"
+
+        val dynamoDbClient = DynamoDbClient.builder()
+            .endpointOverride(URI.create(endpoint))
+            .build()        
+    }
+}
+```
+
+The connection is very straightforward since we don't have to authenticate to connect to our local DynamoDB, the only thing that we have to do is to set our endpoint to be `http://localhost:8000`. With the `dynamoDbClient` now we can proceed to create the table.  
+
+```kotlin
+class DynamoDbTaskRepositoryShould {
+
+    @Test
+    internal fun `add Task to DynamoDB`() {
+        ...
+        dynamoDbClient.createTable { builder ->
+            builder.tableName("tasqui")
+
+            builder.provisionedThroughput { provisionedThroughput ->
+                provisionedThroughput.readCapacityUnits(5)
+                provisionedThroughput.writeCapacityUnits(5)
+            }
+
+            builder.keySchema(
+                KeySchemaElement.builder()
+                    .attributeName("task_id")
+                    .keyType(KeyType.HASH)
+                    .build()
+            )
+
+            builder.attributeDefinitions(
+                AttributeDefinition.builder()
+                    .attributeName("task_id")
+                    .attributeType(ScalarAttributeType.N)
+                    .build()
+            )
+        }
+    }
+}
+```
+
+So, what's going on in this `createTable` method? Let's break down command by command and see:
+
+```kotlin
+builder.tableName("tasqui")
+```
+
+This is a fairly easy part, we are just setting the name of the table, then we have:
+
+```kotlin
+builder.provisionedThroughput { provisionedThroughput ->
+    provisionedThroughput.readCapacityUnits(5)
+    provisionedThroughput.writeCapacityUnits(5)
+}
+```
+
+This part is seeting the throughput for the table, which is the ability to read and write things to the db. We are setting the read and write throughput to 5, but what 5 means exactly? How the throughput is calculated? 
+
+The throughput is measured in `units`, each `unit` might have different values depending in which kind of operation you are doing. For reads, each `unit` is 4Kb/s for consistently strong read, and 8Kb/s for eventually consistent. While writing things get a bit easier, 1 `unit` is 1Kb/s and you don't have any difference between strong or eventual consistency. 
+
+In this case, 5 was chosen since is the default value that Amazon gives to you in the free tier.  
+
```

Comment:
Primary Key is not capitalized consistently (it's in title case here, but lower case in the next paragraph)